projects/plugins/social/src/class-jetpack-social.php

```diff
@@ -299,21 +299,9 @@ public function set_up_sharing_limits() {
 		if ( $this->has_paid_plan() ) {
 			return;
 		}
-		global $publicize;
-
-		$info = $publicize->get_publicize_shares_info( \Jetpack_Options::get_option( 'id' ) );
-
-		if ( is_wp_error( $info ) ) {
-			return;
-		}
 
-		if ( empty( $info['is_share_limit_enabled'] ) ) {
-			return;
-		}
+		global $publicize;
 
-		$connections      = $publicize->get_filtered_connection_data();
-		$shares_remaining = $info['shares_remaining'];
-		$share_limits     = new Automattic\Jetpack\Social\Share_Limits( $connections, $shares_remaining );
-		$share_limits->enforce_share_limits();
+		$publicize->set_up_sharing_limits();
```

Comment:
Should we move the `has_paid_plan` check into this method? We could also add something to make sure it's only run on certain admin requests, although that might need a filter, so we can add the Social admin screen to it.